### PR TITLE
Finalize context menu and update docs

### PR DESCRIPTION
**Pull request by jonas-tai**
### Summary of changes

Update docs and retain attribution highlighting while context menu is open

### Context and reason for change

The context menu was not part of the documentation so far. The highlighting of a hovered attribution disappeared when opening the context menu. This made it unclear, on which attribution the actions of the context menu are executed on. This commit fixes this.

### How can the changes be tested

Read docs and test context menu with test file.

### CODE REVIEW

File: src/Frontend/Components/ContextMenu/ContextMenu.tsx [hunk 1]
```diff
@@ -43,6 +45,8 @@ const BUTTON_TITLE_TO_ICON_MAP: {
   [ButtonText.UnmarkForReplacement]: <CheckBoxIcon fontSize="small" />,
   [ButtonText.ReplaceMarked]: <MergeTypeIcon fontSize="small" />,
   [ButtonText.ShowResources]: <OpenInBrowserIcon fontSize="small" />,
+  [ButtonText.Hide]: <VisibilityIcon fontSize="small" />,
+  [ButtonText.Unhide]: <VisibilityOffIcon fontSize="small" />,
```
Comment (benedikt-richter) [inline src/Frontend/Components/ContextMenu/ContextMenu.tsx:49]: Probalby these two icons should be interchanged. I.e. the eye for "Unhide"

File: src/Frontend/Components/ContextMenu/__tests__/ContextMenu.test.tsx [hunk 2]
```diff
@@ -121,4 +121,31 @@ describe('The ContextMenu', () => {
     fireEvent.click(screen.getByText(ButtonText.Save));
     expect(onClickMock).toHaveBeenCalledTimes(1);
   });
+
+  test('renders and calls onOpen and onClose correctly', () => {
+    const testElementText = 'Test Element';
+    const onCloseMock = jest.fn();
+    const onOpenMock = jest.fn();
+    render(
+      <div>
```
Comment (benedikt-richter) [inline src/Frontend/Components/ContextMenu/__tests__/ContextMenu.test.tsx:130]: why do we have a surrounding div here?

File: src/Frontend/Components/ContextMenu/ContextMenu.tsx [hunk 3]
```diff
@@ -115,13 +127,16 @@ export function ContextMenu(props: ContextMenuProps): ReactElement | null {
...
 
   return displayedMenuItems ? (
-    <>
-      <div
-        onClick={clickHandlers.onClick}
-        onContextMenu={clickHandlers.onContextMenu}
-      >
-        {props.children}
-      </div>
+    <div
```
Comment (benedikt-richter) [inline src/Frontend/Components/ContextMenu/ContextMenu.tsx:130]: why is this moved out of the original div?

File: src/Frontend/Components/ListCard/ListCard.tsx [hunk 4]
```diff
@@ -24,6 +24,10 @@ const useStyles = makeStyles({
       cursor: 'pointer',
     },
   },
+  hoveredPackage: {
+    border: `1px ${OpossumColors.white} solid`,
+    background: OpossumColors.lightestBlueOnHover,
+  },
```
Comment (benedikt-richter) [inline src/Frontend/Components/ListCard/ListCard.tsx:30]: this should be extracted not to duplicate styling (same for other styles on hover)

File: src/Frontend/Components/PackageCard/PackageCard.tsx [hunk 5]
```diff
@@ -310,6 +311,10 @@ export function PackageCard(props: PackageCardProps): ReactElement | null {
         },
       ];
 
+  function updateIsContextMenuOpen(): void {
```
Comment (benedikt-richter) [inline src/Frontend/Components/PackageCard/PackageCard.tsx:314]: could be `toggleIsContextMenuOpen`

File: src/Frontend/test-helpers/context-menu-test-helpers.ts [hunk 6]
```diff
@@ -373,6 +376,13 @@ function openContextMenuOnCardPackageCard(
   fireEvent.contextMenu(screen.getByText(cardLabel) as Element);
 }
 
+function closeContextMenuOnCardPackageCard(
```
Comment (benedikt-richter) [inline src/Frontend/test-helpers/context-menu-test-helpers.ts:379]: why do we need this now and did not need it before?

**Review by benedikt-richter — APPROVED** [review]
nice 👍 
just minor comments

File: src/Frontend/Components/ContextMenu/__tests__/ContextMenu.test.tsx (hunk 2 above)
Comment (jonas-tai) [reply src/Frontend/Components/ContextMenu/__tests__/ContextMenu.test.tsx:130]: Removed it👍

File: src/Frontend/Components/ContextMenu/ContextMenu.tsx (hunk 3 above)
Comment (jonas-tai) [reply src/Frontend/Components/ContextMenu/ContextMenu.tsx:130]: Looks like I forgot to move it back, after I made some experiments. I reverted it👍

File: src/Frontend/test-helpers/context-menu-test-helpers.ts (hunk 6 above)
Comment (jonas-tai) [reply src/Frontend/test-helpers/context-menu-test-helpers.ts:379]: This was actually a bug. I noticed that some of the tests failed, due to the context menu still being open. I think most integration tests just closed it coincidentally by executing other actions.

**Comment by jonas-tai** [comment]
> nice 👍 just minor comments

I implemented the comments👍

**Review by nicarl — APPROVED** [review]
Something to discuss in the team: Do we want to get rid of the buttons eventually?
